src/state_machine.zig

```diff
@@ -1890,14 +1948,16 @@ pub fn StateMachineType(
                 .cr_account = &cr_account_new,
             });
 
-            if (t.timeout > 0) {
-                const expires_at = t.timestamp + t.timeout_ns();
+            if (t2.timeout > 0) {
+                assert(t2.flags.pending);
+                assert(!t2.flags.imported);
+                const expires_at = t2.timestamp + t2.timeout_ns();
                 if (expires_at < self.expire_pending_transfers.pulse_next_timestamp) {
                     self.expire_pending_transfers.pulse_next_timestamp = expires_at;
                 }
             }
 
-            self.commit_timestamp = t.timestamp;
+            self.commit_timestamp = t2.timestamp;
```

Comment:
Yes, this is just a minor refactor. Currently only `amount` differs, but since we are inserting `t2`, I think it makes sense to place all validations there, avoiding possible misbehavior if something changes other fields in the future.